st2common/st2common/services/packs.py

```diff
@@ -106,7 +114,11 @@ def _fetch_and_compile_index(index_urls, logger=None, proxy_config=None):
         index_json = None
 
         try:
-            request = requests.get(index_url, proxies=proxies_dict, verify=verify)
+            request = requests.get(
+                index_url,
+                proxies=proxies_dict if not bypass_proxy else None,
+                verify=verify if not bypass_proxy else None
```

Comment:
I understand the need of overriding the proxies, but in this case, why it's needed to disable the `verify` flag?

Wouldn't it downgrade the security for the users?
 https://github.com/StackStorm/st2/blob/8ad5e5dcc4f7fa14c6ab5f1aae5faf240f389d07/st2common/st2common/services/packs.py#L88-L92
it's better to allow them to pass `proxy_ca_bundle_path` instead of disabling the cert verification.

Reply:
I'll have to look at this again. Verify accepts a bool or a string to the path of the ca bundle to use. I was thinking more in terms of bypassing the CA cert bundle string than the bool. I'll look at it later today and fix it. Thanks for pointing it out.